src/http-gateways/trustless-gateway.md

```diff
@@ -301,6 +302,112 @@ of their presence in the DAG or the value assigned to the "dups" parameter, as
 the raw data is already present in the parent block that links to the identity
 CID.
 
+## `meta` (content type parameter)
+
+The `meta` parameter allows clients to request the server to include additional metadata about the CAR along with the response body.
+
+The value of this parameter includes both the location where the metadata is given (e.g. `eof`) as well as the type of data received (e.g. `json`) separated by a `+`, to give a value such as `meta=eof+json`
+
+When the location parameter is set to `eof`, which is currently the only supported value, the server SHOULD respond with the following response body:
+
+```
+<Response body as CARv1 stream> <0x00 byte> <Metadata>
+```
+
+The only supported value for the data type parameter is `json`. This signifies that the metadata MUST be a JSON object.
+
+This parameter MUST only be used with CAR `version=1`.
+
+When the parameter is not set or does not equal `eof+json`, the server SHOULD not add any extra blocks to the response, neither the 0x00 byte nor any metadata.
+
+When `meta=eof+json`, the JSON object SHOULD conform to the following [JSON schema](https://json-schema.org/).
```

Comment:
Discussion points:

- In the current spec & IPIP, we are formatting metadata as JSON. Should we say DAG-JSON instead?
- Do we want to serialise the metadata as a CAR block, prefixing the JSON data with `varint | CID` header?

Reply:
@rvagg  thoughts? Value added in DAG-JSON prefixed with own CID is that it allows client to detect truncation beyond `0x00` byte.

Reply:
I believe clients can already easily detect truncation of the metadata block.
- The block is a DAG-JSON object, it must start with `{` and end with a matching `}`.
- If the block is truncated, it will not end with the matching `}` and the JSON parser will throw an error.